src/transformers/models/fast/modeling_fast.py

```diff
@@ -0,0 +1,380 @@
+
+
+def get_same_padding(kernel_size):
+    if isinstance(kernel_size, tuple):
+        padding1 = get_same_padding(kernel_size[0])
+        padding2 = get_same_padding(kernel_size[1])
+        return padding1, padding2
+    return kernel_size // 2
+
+
+class FastRepConvLayer(nn.Module):
+    def __init__(
+        self, in_channels: int, out_channels: int, kernel_size: Tuple[int, int], stride: int = 1, bias: bool = False
+    ) -> None:
+        super().__init__()
+
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.kernel_size = kernel_size
+        self.stride = stride
+
+        padding = (int((kernel_size[0] - 1) / 2), int((kernel_size[1] - 1) / 2))
+
+        self.activation = nn.ReLU(inplace=True)
+
+        self.main_conv = nn.Conv2d(
+            in_channels=in_channels,
+            out_channels=out_channels,
+            kernel_size=kernel_size,
+            stride=stride,
+            padding=padding,
+            bias=bias,
+        )
+        self.main_batch_norm = nn.BatchNorm2d(num_features=out_channels)
+
+        vertical_padding = (int((kernel_size[0] - 1) / 2), 0)
+        horizontal_padding = (0, int((kernel_size[1] - 1) / 2))
+
+        if kernel_size[1] != 1:
+            self.vertical_conv = nn.Conv2d(
+                in_channels=in_channels,
+                out_channels=out_channels,
+                kernel_size=(kernel_size[0], 1),
+                stride=stride,
+                padding=vertical_padding,
+                bias=bias,
+            )
+            self.vertical_batch_norm = nn.BatchNorm2d(num_features=out_channels)
+        else:
+            self.vertical_conv = nn.Identity()
+            self.vertical_batch_norm = nn.Identity()
+
+        if kernel_size[0] != 1:
+            self.horizontal_conv = nn.Conv2d(
+                in_channels=in_channels,
+                out_channels=out_channels,
+                kernel_size=(1, kernel_size[1]),
+                stride=stride,
+                padding=horizontal_padding,
+                bias=bias,
+            )
+            self.horizontal_batch_norm = nn.BatchNorm2d(num_features=out_channels)
+        else:
+            self.horizontal_conv = nn.Identity()
+            self.horizontal_batch_norm = nn.Identity()
+
+        # TODO: check if needed
+        self.rbr_identity = nn.BatchNorm2d(out_channels) if out_channels == in_channels and stride == 1 else None
+
+    def forward(self, hidden_states: torch.Tensor):
+        # if self.training:
+        main = self.main_conv(hidden_states)
+        main = self.main_batch_norm(main)
+
+        vertical = self.vertical_conv(hidden_states)
+        vertical = self.vertical_batch_norm(vertical)
+
+        horizontal = self.horizontal_conv(hidden_states)
+        horizontal = self.horizontal_batch_norm(horizontal)
+
+        if self.rbr_identity is not None:
+            rbr_identity = self.rbr_identity(hidden_states)
+        else:
+            rbr_identity = 0
```

Comment:
Should be the same as other modules, resolved in init
```suggestion
        identity = self.identity(hidden_states)

        if self.add_identity:
            hidden_states = hidden_states + identity

        return self.activation(main + vertical + horizontal)
```

Reply:
indeed, i did some workaround to fix it